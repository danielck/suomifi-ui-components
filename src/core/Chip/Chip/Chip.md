The `<Chip>` component is used when you want to present concise information in a clear visually prominent way, for example to display the selected values of a listbox. In its most typical usecase, the Chip is a button with a "remove" icon.

Examples:

- [Basic use](./#/Components/Chip?id=basic-use)
- [Without remove icon](./#/Components/Chip?id=without-remove-icon)
- [Disabled](./#/Components/Chip?id=disabled)
- [Static chip](./#/Components/Chip?id=static-chip)

<div style="margin-bottom: 5px">
  [Props & methods (Chip)](./#/Components/Chip?id=props--methods)
</div>
<div style="margin-bottom: 40px">
  [Props & methods (StaticChip)](./#/Components/Chip?id=staticchip)
</div>

### Basic use

It is important to set the `actionLabel` text so that can describe the function of the Chip.

```js
import { Chip } from 'suomifi-ui-components';

<Chip
  removable
  actionLabel="Deselect item"
  onClick={() => console.log('Deselected: Finland')}
>
  Finland
</Chip>;
```

### Without remove icon

The remove icon can be hidden to allow for more use cases

```js
import { Chip } from 'suomifi-ui-components';

<Chip
  actionLabel="Log item to console"
  onClick={() => console.log('Clicked option: Finland')}
>
  Finland
</Chip>;
```

### Disabled

In the example below

- The first Chip is fully `disabled`
- The second one is `aria-disabled`. Aria-disabled allows focus but disables the Chip's `onClick()` functionality.

```js
import { Chip } from 'suomifi-ui-components';

<>
  <Chip
    disabled
    removable
    actionLabel="Deselect item"
    onClick={() => console.log('Deselected: Finland')}
  >
    Finland
  </Chip>

  <Chip
    aria-disabled
    removable
    actionLabel="Deselect item"
    onClick={() => console.log('Deselected: Finland')}
  >
    Finland
  </Chip>
</>;
```

### Static chip

`<StaticChip>` is a Chip without the button functionality, i.e only for visual presentation.

```js
import { StaticChip } from 'suomifi-ui-components';

<StaticChip>Finland</StaticChip>;
```

### Props & methods
