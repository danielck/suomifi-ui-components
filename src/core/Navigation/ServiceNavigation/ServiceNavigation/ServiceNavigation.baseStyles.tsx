import { font } from '../../../theme/reset';
import { css } from 'styled-components';
import { SuomifiTheme } from '../../../theme';

export const baseStyles = (theme: SuomifiTheme) => css`
  ${font(theme)('bodyText')}
  &.fi-service-navigation {
    & .fi-service-navigation_list {
      list-style-type: none;
      margin: 0;
      padding: 0;
    }
  }
`;
