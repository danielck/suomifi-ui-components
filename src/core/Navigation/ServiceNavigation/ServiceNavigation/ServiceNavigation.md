### Basic usage

Please use `<RouterLink>` as the child of `<ServiceNavigationItem>` to get inteded CSS styles. `<RouterLink>` is polymorphic, and can be rendered as any component of your choice, for example React Router Link.

```js
import {
  ServiceNavigation,
  ServiceNavigationItem,
  RouterLink,
  ExternalLink,
  StaticChip
} from 'suomifi-ui-components';

const CustomButton = (props) => {
  const { children, ...passProps } = props;
  return <button {...passProps}>{props.children}</button>;
};

<div style={{ width: '300px' }}>
  <ServiceNavigation>
    <ServiceNavigationItem>
      <RouterLink href="/" aria-label="Inbox. 16 unread messages.">
        Inbox
        <StaticChip style={{ marginLeft: '15px' }} aria-hidden>
          16
        </StaticChip>
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={ExternalLink}
        href="https://suomi.fi"
        hideIcon
      >
        Sent
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem selected>
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
        aria-current="page"
      >
        New Message
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
      >
        Drafts
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={CustomButton}
        onClick={() => console.log('Nav item clicked')}
      >
        Settings
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem disabled>
      <RouterLink>Devices</RouterLink>
    </ServiceNavigationItem>
  </ServiceNavigation>
</div>;
```

For frameworks where its internal link component is used as a wrapper for the actual link, for example NextJS, the following approach can be used:

```jsx static
<ServiceNavigationItem>
  <NextJSLink href="/some-route" passHref>
    <RouterLink>Inbox</RouterLink>
  </NextJSLink>
</ServiceNavigationItem>
```

### Small screen variant

```js
import {
  ServiceNavigation,
  ServiceNavigationItem,
  RouterLink,
  ExternalLink,
  StaticChip
} from 'suomifi-ui-components';

const Comp = (props) => {
  const { children, ...passProps } = props;
  return <div {...passProps}>{props.children}</div>;
};

const expandButtonContent = (
  <>
    <span>DRAFTS</span>
    <StaticChip style={{ marginLeft: '15px' }} aria-hidden>
      3
    </StaticChip>
  </>
);

<div style={{ width: '300px' }}>
  <ServiceNavigation
    variant="smallScreen"
    smallScreenExpandButtonText={expandButtonContent}
    initiallyExpanded={false}
  >
    <ServiceNavigationItem>
      <RouterLink href="/">Inbox</RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={ExternalLink}
        href="https://suomi.fi"
        hideIcon
      >
        Sent
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
      >
        New Message
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem selected>
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
        aria-current="page"
        aria-label="Drafts. 3 drafts waiting."
      >
        Drafts
        <StaticChip style={{ marginLeft: '15px' }} aria-hidden>
          3
        </StaticChip>
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={Comp}
        onClick={() => console.log('Nav item clicked')}
        role="button"
        tabIndex="0"
      >
        Settings
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem disabled>
      <RouterLink>Devices</RouterLink>
    </ServiceNavigationItem>
  </ServiceNavigation>
</div>;
```
