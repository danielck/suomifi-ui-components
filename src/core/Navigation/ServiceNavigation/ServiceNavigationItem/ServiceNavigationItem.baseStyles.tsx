import { font } from '../../../theme/reset';
import { css } from 'styled-components';
import { SuomifiTheme } from '../../../theme';

export const baseStyles = (theme: SuomifiTheme) => css`
  ${font(theme)('bodySemiBold')};

  /* stylelint-disable no-descending-specificity */
  /* Nested :hover etc selectors do not work well with this rule. */
  &.fi-service-navigation-item {
    display: flex;
    align-items: center;

    &:hover,
    &:active {
      background: ${theme.colors.depthSecondary};
      cursor: pointer;
    }

    .fi-link--router {
      text-decoration: none;
      color: ${theme.colors.blackBase};
      display: flex;
      padding: ${theme.spacing.xs} ${theme.spacing.m};
      text-transform: uppercase;
      flex: 1;
      cursor: pointer;
      background: inherit;
      border: none;

      &:hover,
      &:active,
      &:visited {
        text-decoration: none;
        color: ${theme.colors.blackBase};
      }
    }

    &--disabled {
      .fi-link--router {
        color: ${theme.colors.depthBase};
        cursor: not-allowed;

        &:hover,
        &:active,
        &:visited {
          color: ${theme.colors.depthBase};
        }
      }
    }

    &--selected {
      display: flex;
      align-items: center;
      background: ${theme.colors.depthSecondary};
      border-left: 4px solid ${theme.colors.highlightBase};

      .fi-link--router {
        font-weight: bold;
        text-decoration: none;
        color: ${theme.colors.blackBase};
        display: flex;
        padding: ${theme.spacing.xs} ${theme.spacing.l};
        padding-left: calc(${theme.spacing.m} - 4px);
        text-transform: uppercase;
        flex: 1;

        &:hover,
        &:active,
        &:visited {
          text-decoration: none;
          color: ${theme.colors.blackBase};
        }
      }
    }
  }
`;

/* stylelint-enable no-descending-specificity */
