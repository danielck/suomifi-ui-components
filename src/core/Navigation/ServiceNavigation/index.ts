export {
  ServiceNavigation,
  ServiceNavigationProps,
} from './ServiceNavigation/ServiceNavigation';
export {
  ServiceNavigationItem,
  ServiceNavigationItemProps,
} from './ServiceNavigationItem/ServiceNavigationItem';
