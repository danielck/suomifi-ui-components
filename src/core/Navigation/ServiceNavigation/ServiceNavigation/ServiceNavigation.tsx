import React, { ReactNode, useState } from 'react';
import {
  HtmlButton,
  HtmlDiv,
  HtmlNav,
  HtmlSpan,
  HtmlUl,
} from '../../../../reset';
import styled from 'styled-components';
import { SuomifiThemeConsumer } from '../../../theme';
import { baseStyles } from './ServiceNavigation.baseStyles';
import classnames from 'classnames';
import { Icon } from '../../../Icon/Icon';

export interface ServiceNavigationProps {
  children: ReactNode | ReactNode[];
  /**
   * Normal or small screen variant
   * @default normal
   */
  variant?: 'default' | 'smallScreen';
  /**
   * Whether the menu is initially expanded. Only applies to smallScreen variant.
   * @default true
   */
  initiallyExpanded?: boolean;
  /** When using smallScreen variant, this is the text showed in the expander button.
   * Intended to be the same text as the currently selected nav item.  */
  smallScreenExpandButtonText?: string | ReactNode;
  /** Custom classname to extend or customize */
  className?: string;
}

const baseClassName = 'fi-service-navigation';
const smallScreenClassName = `${baseClassName}--small-screen`;
const listClassName = `${baseClassName}_list`;
const smallScreenExpandButtonClassName = `${baseClassName}_expand-button`;

const BaseServiceNavigation = ({
  children,
  variant,
  initiallyExpanded,
  smallScreenExpandButtonText,
  className,
}: ServiceNavigationProps) => {
  const initiallyExpandedValue =
    initiallyExpanded !== undefined ? initiallyExpanded : true;
  const [smallScreenNavOpen, setSmallScreenNavOpen] = useState(
    initiallyExpandedValue,
  );
  return (
    <HtmlDiv
      className={classnames(baseClassName, className, {
        [smallScreenClassName]: variant === 'smallScreen',
      })}
    >
      {variant === 'smallScreen' && (
        <HtmlButton
          className={smallScreenExpandButtonClassName}
          onClick={() => setSmallScreenNavOpen(!smallScreenNavOpen)}
          aria-expanded={smallScreenNavOpen}
        >
          <HtmlSpan
            style={{
              display: 'flex',
              justifyContent: 'center',
              alignItems: 'center',
            }}
          >
            {smallScreenExpandButtonText}
          </HtmlSpan>
          <Icon icon={smallScreenNavOpen ? 'chevronDown' : 'chevronRight'} />
        </HtmlButton>
      )}
      {((variant === 'smallScreen' && smallScreenNavOpen) ||
        variant !== 'smallScreen') && (
        <HtmlNav>
          <HtmlUl className={listClassName}>{children}</HtmlUl>
        </HtmlNav>
      )}
    </HtmlDiv>
  );
};

const StyledServiceNavigation = styled(BaseServiceNavigation)`
  ${({ theme }) => baseStyles(theme)}
`;

const ServiceNavigation = (props: ServiceNavigationProps) => (
  <SuomifiThemeConsumer>
    {({ suomifiTheme }) => (
      <StyledServiceNavigation theme={suomifiTheme} {...props} />
    )}
  </SuomifiThemeConsumer>
);

ServiceNavigation.displayName = 'ServiceNavigation';
export { ServiceNavigation };
