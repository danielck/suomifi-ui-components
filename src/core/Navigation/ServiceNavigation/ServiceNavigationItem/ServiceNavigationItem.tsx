import React, { ReactNode } from 'react';
import classnames from 'classnames';
import { HtmlLi } from '../../../../reset';
import { SuomifiThemeConsumer, SuomifiThemeProp } from '../../../theme';
import { baseStyles } from './ServiceNavigationItem.baseStyles';
import styled from 'styled-components';

interface BasicServiceNavigationItemProps {
  /** Custom class */
  className?: string;
  /** Use the polymorphic RouterLink component as child to get intended CSS styling */
  children: ReactNode;
  /** Disable the item */
  disabled?: boolean;
}

type SeletedProps =
  | {
      /** Show item as the selected one */
      selected: boolean;
      /** Selected item information for screen reader. Required when `selected` is true */
      ariaCurrent: 'step' | 'page' | 'location';
    }
  | {
      selected?: never;
      ariaCurrent?: never;
    };

export type ServiceNavigationItemProps = BasicServiceNavigationItemProps &
  SeletedProps;

const baseClassName = 'fi-service-navigation-item';
const selectedClassName = `${baseClassName}--selected`;
const disabledClassName = `${baseClassName}--disabled`;

const BaseServiceNavigationItem = ({
  selected,
  ariaCurrent,
  className,
  children,
  disabled,
  ...passProps
}: ServiceNavigationItemProps) => (
  <HtmlLi
    className={classnames(className, {
      [baseClassName]: !selected,
      [selectedClassName]: selected,
      [disabledClassName]: disabled,
    })}
    aria-current={!!ariaCurrent ? ariaCurrent : undefined}
    aria-disabled={disabled}
    {...passProps}
  >
    {children}
  </HtmlLi>
);

const StyledServiceNavigationItem = styled(
  (props: ServiceNavigationItemProps & SuomifiThemeProp) => {
    const { theme, ...passProps } = props;
    return <BaseServiceNavigationItem {...passProps} />;
  },
)`
  ${({ theme }) => baseStyles(theme)}
`;

const ServiceNavigationItem = (props: ServiceNavigationItemProps) => (
  <SuomifiThemeConsumer>
    {({ suomifiTheme }) => (
      <StyledServiceNavigationItem theme={suomifiTheme} {...props} />
    )}
  </SuomifiThemeConsumer>
);

ServiceNavigationItem.displayName = 'ServiceNavigationItem';
export { ServiceNavigationItem };
