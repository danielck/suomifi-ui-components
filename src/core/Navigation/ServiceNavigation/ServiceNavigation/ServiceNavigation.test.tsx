import React from 'react';
import { render } from '@testing-library/react';
import { axeTest } from '../../../../utils/test';

import { ServiceNavigation } from './ServiceNavigation';
import { ServiceNavigationItem } from '../ServiceNavigationItem/ServiceNavigationItem';
import { ExternalLink, RouterLink } from '../../../Link';
import { StaticChip } from '../../../Chip';

interface TestProps {
  children: string;
  onClick?: () => void;
  role?: string;
  tabIndex?: number;
}

const Comp = ({ children, ...passProps }: TestProps) => (
  <div {...passProps}>{children}</div>
);

const TestServiceNavigation = (
  <ServiceNavigation>
    <ServiceNavigationItem aria-label="16 unread messages">
      <RouterLink
        asComponent={Comp}
        onClick={() => console.log('Nav item clicked')}
        role="button"
        tabIndex={0}
      >
        Inbox
        <StaticChip style={{ marginLeft: '15px' }}>16</StaticChip>
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink asComponent={ExternalLink} href="https://suomi.fi" hideIcon>
        Sent
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem selected ariaCurrent="page">
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
      >
        New Message
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
      >
        Drafts
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink
        asComponent={ExternalLink}
        href="https://www.suomi.fi"
        hideIcon
      >
        Settings
      </RouterLink>
    </ServiceNavigationItem>
    <ServiceNavigationItem>
      <RouterLink>Devices</RouterLink>
    </ServiceNavigationItem>
  </ServiceNavigation>
);

test('calling render with the same component on the same container does not remount', () => {
  const ParagraphRendered = render(TestServiceNavigation);
  const { container } = ParagraphRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test(
  'should not have basic accessibility issues',
  axeTest(TestServiceNavigation),
);
