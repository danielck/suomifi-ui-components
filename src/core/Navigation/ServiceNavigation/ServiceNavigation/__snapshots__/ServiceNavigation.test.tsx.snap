// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.c9 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  background-color: transparent;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9::before,
.c9::after {
  box-sizing: border-box;
}

.c9:hover,
.c9:active,
.c9:focus,
.c9:focus-within {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: list-item;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c7 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c7::before,
.c7::after {
  box-sizing: border-box;
}

.c3 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  list-style-type: decimal;
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 0;
  margin-right: 0;
  padding-left: 40px;
}

.c3::before,
.c3::after {
  box-sizing: border-box;
}

.c1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1.fi-service-navigation--small-screen {
  background: hsl(212,63%,95%);
}

.c1.fi-service-navigation--small-screen .fi-service-navigation_expand-button {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  border: 1px solid hsl(212,63%,90%);
}

.c1.fi-service-navigation--small-screen .fi-service-navigation_expand-button .fi-icon {
  height: 25px;
  width: 25px;
}

.c1.fi-service-navigation .fi-service-navigation_list {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
}

.c5.fi-service-navigation-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5.fi-service-navigation-item:hover,
.c5.fi-service-navigation-item:active {
  background: hsl(215,100%,97%);
  cursor: pointer;
}

.c5.fi-service-navigation-item .fi-link--router {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,16%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 20px;
  text-transform: uppercase;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
  background: inherit;
  border: none;
}

.c5.fi-service-navigation-item .fi-link--router:hover,
.c5.fi-service-navigation-item .fi-link--router:active,
.c5.fi-service-navigation-item .fi-link--router:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,16%);
}

.c5.fi-service-navigation-item--disabled .fi-link--router {
  color: hsl(202,7%,67%);
  cursor: not-allowed;
}

.c5.fi-service-navigation-item--disabled .fi-link--router:hover,
.c5.fi-service-navigation-item--disabled .fi-link--router:active,
.c5.fi-service-navigation-item--disabled .fi-link--router:visited {
  color: hsl(202,7%,67%);
}

.c5.fi-service-navigation-item--selected {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsl(215,100%,97%);
  border-left: 4px solid hsl(212,63%,45%);
}

.c5.fi-service-navigation-item--selected .fi-link--router {
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,16%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 25px;
  padding-left: calc(20px - 4px);
  text-transform: uppercase;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5.fi-service-navigation-item--selected .fi-link--router:hover,
.c5.fi-service-navigation-item--selected .fi-link--router:active,
.c5.fi-service-navigation-item--selected .fi-link--router:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,16%);
}

.c11 {
  position: absolute;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
}

.c10 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c10.fi-link {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10.fi-link:hover,
.c10.fi-link:active,
.c10.fi-link:focus,
.c10.fi-link:focus-within {
  color: hsl(212,63%,45%);
}

.c10.fi-link:focus,
.c10.fi-link:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10.fi-link:focus {
  outline: 0;
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
}

.c10.fi-link:hover,
.c10.fi-link:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10.fi-link:visited {
  color: hsl(284,36%,45%);
}

.c10 .fi-link_icon {
  padding-left: 4px;
  font-size: 16px;
  box-sizing: content-box;
}

.c6 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c6.fi-link--router {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6.fi-link--router:hover,
.c6.fi-link--router:active,
.c6.fi-link--router:focus,
.c6.fi-link--router:focus-within {
  color: hsl(212,63%,45%);
}

.c6.fi-link--router:focus,
.c6.fi-link--router:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6.fi-link--router:focus {
  outline: 0;
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
}

.c6.fi-link--router:hover,
.c6.fi-link--router:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6.fi-link--router:visited {
  color: hsl(284,36%,45%);
}

.c8 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
}

.c8.fi-chip {
  border-radius: 14px;
  padding: 2px 10px;
  color: hsl(0,0%,100%);
  background: hsl(212,63%,45%);
  max-height: 100%;
  display: inline-block;
}

.c8.fi-chip:focus {
  outline: 0;
  position: relative;
}

.c8.fi-chip:focus::after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
  border-radius: 16px;
}

.c8.fi-chip .fi-chip--content {
  display: block;
  word-break: break-word;
  overflow: hidden;
  line-height: 1.5em;
}

.c8.fi-chip--disabled.fi-chip {
  cursor: not-allowed;
  background: hsl(202,7%,67%);
}

.c8.fi-chip--disabled.fi-chip:hover,
.c8.fi-chip--disabled.fi-chip:active {
  background: hsl(202,7%,67%);
}

<div
  class="c0 fi-service-navigation c1"
>
  <nav
    class="c2"
  >
    <ul
      class="c3 fi-service-navigation_list"
    >
      <li
        aria-label="16 unread messages"
        class="c4 c5 fi-service-navigation-item"
      >
        <div
          class="fi-link--router c6"
          role="button"
          tabindex="0"
        >
          Inbox
          <span
            class="c7 fi-chip c8"
            style="margin-left: 15px;"
          >
            <span
              class="c7 fi-chip--content"
            >
              16
            </span>
          </span>
        </div>
      </li>
      <li
        class="c4 c5 fi-service-navigation-item"
      >
        <a
          class="c9 fi-link c10 fi-link--router c6 fi-link--external"
          href="https://suomi.fi"
          rel="noopener"
          target="_blank"
        >
          Sent
          <span
            class="c7 c11 fi-visually-hidden"
          />
        </a>
      </li>
      <li
        aria-current="page"
        class="c4 c5 fi-service-navigation-item--selected"
      >
        <a
          class="c9 fi-link c10 fi-link--router c6 fi-link--external"
          href="https://www.suomi.fi"
          rel="noopener"
          target="_blank"
        >
          New Message
          <span
            class="c7 c11 fi-visually-hidden"
          />
        </a>
      </li>
      <li
        class="c4 c5 fi-service-navigation-item"
      >
        <a
          class="c9 fi-link c10 fi-link--router c6 fi-link--external"
          href="https://www.suomi.fi"
          rel="noopener"
          target="_blank"
        >
          Drafts
          <span
            class="c7 c11 fi-visually-hidden"
          />
        </a>
      </li>
      <li
        class="c4 c5 fi-service-navigation-item"
      >
        <a
          class="c9 fi-link c10 fi-link--router c6 fi-link--external"
          href="https://www.suomi.fi"
          rel="noopener"
          target="_blank"
        >
          Settings
          <span
            class="c7 c11 fi-visually-hidden"
          />
        </a>
      </li>
      <li
        class="c4 c5 fi-service-navigation-item"
      >
        <a
          class="c9 fi-link--router c6"
        >
          Devices
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
